pkg/kubelet/dockertools/manager.go

```diff
@@ -1925,13 +1919,30 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, pod
 			}
 		}
 
+		labels := map[string]string{}
+		containerStatuses := podStatus.ContainerStatuses
+		// podStatus is generated by GetPodStatus(). In GetPodStatus(), we make sure that ContainerStatuses
+		// contains statuses of all containers in pod.Spec.Containers.
+		// ContainerToStart is a subset of pod.Spec.Containers, we should always find a result here.
+		// (taotaotheripper) Which is better, sort.Search or for loop or create a map first?
+		containerStatusesLength := len(containerStatuses)
+		containerStatusIdx := sort.Search(containerStatusesLength, func(i int) bool { return containerStatuses[i].Name >= container.Name })
+		if containerStatusIdx == containerStatusesLength {
+			// TODO (taotaotheripper) However, it hits here in integration test, figure out why.
+			// Can not found such a container
+			labels[kubernetesContainerRestartCountLabel] = "0"
+		} else {
+			containerStatus := containerStatuses[containerStatusIdx]
+			labels[kubernetesContainerRestartCountLabel] = strconv.Itoa(containerStatus.RestartCount + 1)
```

Comment:
As mentioned in comment above, if the status is terminated or `LastTerminationState` is non-nil, do `RestartCount +1`; otherwise, assume this is a new container and use 0 restart count.